ngraph/test/util/engine/engine_traits.hpp

```diff
@@ -0,0 +1,29 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+namespace ngraph
+{
+    namespace test
+    {
+        /// A specialization of this struct should be created for each engine type
+        template <typename Engine>
+        struct EngineTraits
```

Comment:
This name is too generic IMHO. Moreover I don't know what to expect to be in here?

Reply:
Maybe more verbose idea would be to have traits like: 
```
template <typename Engine>
struct supports_devices
{
  static constexpr bool value = false;
};

template <typename Engine>
struct supports_dynamic
{
  static constexpr bool value = false;
};
```
And `device` you could define as static constexpr function within each Engine class.

Reply:
I'm ok with renaming it(is `TestEngineTraits` a better name?) but I can't switch to those 2 separate classes. I need this particular form so that overload resolution works the way I need it in this line https://github.com/openvinotoolkit/openvino/pull/934/files#diff-5b3baf93c1bb54cd6dbaef3b48745410R68